src/test/java/io/jenkins/plugins/analysis/warnings/FlexiblePublishITest.java

```diff
@@ -0,0 +1,143 @@
+import io.jenkins.plugins.analysis.warnings.recorder.pageobj.IssueRow;
+import io.jenkins.plugins.analysis.warnings.recorder.pageobj.IssuesTable;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Test the flexible publish plugin in combination with the warnings-ng-plugin.
+ *
+ * @author Tobias Redl
+ * @author Andreas Neumeier
+ */
+public class FlexiblePublishITest extends IntegrationTestWithJenkinsPerSuite {
+    private static final String JAVA_FILE = "java2Warnings.txt";
+    private static final String JAVA_FILE2 = "java-start.txt";
+    private static final String JAVA_PATTERN = "**/*.txt";
+
+    /**
+     * Test that the same tool can be used twice with different configuration.
+     */
+    @Test
+    public void shouldAnalyseJavaTwice() {
+        FreeStyleProject project = createFreeStyleProject();
+        copySingleFileToWorkspace(project, JAVA_FILE, JAVA_FILE);
+        copySingleFileToWorkspace(project, JAVA_FILE2, JAVA_FILE2);
+
+        final String toolId = "hm-edu1";
+        final String toolId2 = "hm-edu2";
+
+        IssuesRecorder publisher = constructJavaIssuesRecorder(JAVA_FILE, toolId, false, 1, 9);
+        IssuesRecorder publisher2 = constructJavaIssuesRecorder(JAVA_PATTERN, toolId2, false, 1, 3);
+
+        project.getPublishersList().add(new FlexiblePublisher(Arrays.asList(
+                constructConditionalPublisher(publisher),
+                constructConditionalPublisher(publisher2)
+        )));
+
+        AnalysisResult analysisResult = scheduleSuccessfulBuild(project);
+        assertThat(analysisResult.isSuccessful()).isEqualTo(true);
+        HealthReport healthReport = project.getBuildHealth();
+        assertThat(healthReport.getScore()).isEqualTo(0);
+        checkDetailsViewForIssues(project, analysisResult, toolId, 2);
+        checkDetailsViewForIssues(project, analysisResult, toolId2, 4);
+    }
+
+    /**
+     * Test that the same tool can be used twice with different configuration and is aggregated afterwards.
+     */
+    @Test
+    public void shouldAnalyseJavaWithAggregation() {
+        FreeStyleProject project = createFreeStyleProject();
+        copySingleFileToWorkspace(project, JAVA_FILE, JAVA_FILE);
+        copySingleFileToWorkspace(project, JAVA_FILE2, JAVA_FILE2);
+
+        final String toolId = "hm-edu1";
+        final String toolId2 = "hm-edu2";
+
+        IssuesRecorder publisher = constructJavaIssuesRecorder(JAVA_FILE, toolId, true);
+        IssuesRecorder publisher2 = constructJavaIssuesRecorder(JAVA_PATTERN, toolId2, true);
+
+        project.getPublishersList().add(new FlexiblePublisher(Arrays.asList(
+                constructConditionalPublisher(publisher),
+                constructConditionalPublisher(publisher2)
+        )));
+
+        AnalysisResult analysisResult = scheduleSuccessfulBuild(project);
+        assertThat(analysisResult.isSuccessful()).isEqualTo(true);
+        //is there a bug with the aggregation?
```

Comment:
We would assume, that all the publishers will be shown as one, but they aren't.

![image](https://user-images.githubusercontent.com/15648487/58643153-098cbb00-82ff-11e9-80c7-e1fe72b116aa.png)

Reply:
@uhafner Are we right with our assumption?